airflow/models.py

```diff
@@ -336,7 +336,8 @@ def kill_zombies(self, session):
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = False
```

Comment:
yeah, I too found it strange as in `models.py` we are setting `test_mode` to `False`. That's why I added it in PR and left a note, so someone may help in understand this behaviour. Below is the stacktrace of scheduler's `kill_zombies` function:

> [2016-08-11 12:48:19,313] {models.py:313} INFO - Finding 'running' jobs without a recent heartbeat
[2016-08-11 12:48:19,314] {models.py:319} INFO - Failing jobs without heartbeat after 2016-08-11 12:46:04.314176
[2016-08-11 12:48:19,321] {jobs.py:793} ERROR - 'TaskInstance' object has no attribute 'test_mode'
Traceback (most recent call last):
  File "/Users/sumit/src/airflow-sumit/airflow/jobs.py", line 791, in _execute
    dagbag.kill_zombies()
  File "/Users/sumit/src/airflow-sumit/airflow/utils/db.py", line 53, in wrapper
    result = func(*args, **kwargs)
  File "/Users/sumit/src/airflow-sumit/airflow/models.py", line 341, in kill_zombies
    False, ti.get_template_context())
  File "/Users/sumit/src/airflow-sumit/airflow/utils/db.py", line 53, in wrapper
    result = func(*args, **kwargs)
  File "/Users/sumit/src/airflow-sumit/airflow/models.py", line 1516, in get_template_context
    'test_mode': self.test_mode,
AttributeError: 'TaskInstance' object has no attribute 'test_mode'